src/3D/Camera.cpp

```diff
@@ -40,22 +45,32 @@ void Camera::SetProjectionMatrixPerspective(float xFov, float aspect, float near
 glm::vec3 Camera::GetForward() const
 {
 	// Forward is +1 in openblack but is -1 in OpenGL
-	glm::mat3 mRotation = glm::transpose(getRotationMatrix());
+	glm::mat3 mRotation = glm::transpose(GetRotationMatrix());
 	return mRotation * glm::vec3(0, 0, 1);
 }
 
 glm::vec3 Camera::GetRight() const
 {
-	glm::mat3 mRotation = glm::transpose(getRotationMatrix());
+	glm::mat3 mRotation = glm::transpose(GetRotationMatrix());
 	return mRotation * glm::vec3(1, 0, 0);
 }
 
 glm::vec3 Camera::GetUp() const
 {
-	glm::mat3 mRotation = glm::transpose(getRotationMatrix());
+	glm::mat3 mRotation = glm::transpose(GetRotationMatrix());
 	return mRotation * glm::vec3(0, 1, 0);
 }
 
+float Camera::GetGroundHeight() const
+{
+	float intersectDistance = 0.0f;
+	const auto rayDirection = glm::vec3(0.0f, -1.0f, 0.0f);
+	const auto plane_origin = glm::vec3(0.0f, 0.0f, 0.0f);
+	const auto plane_normal = glm::vec3(0.0f, 1.0f, 0.0f);
+	glm::intersectRayPlane(_position, rayDirection, plane_origin, plane_normal, intersectDistance);
+	return intersectDistance;
+}
```

Comment:
Ground height is in function of the land heighmap. I thought we recently implemented this using bullet and intersection with the actual geometric.
This virtual plane is definitely wrong for getting the height.
If you mean to get the altitude, the function name should reflect it better.

Reply:
this function is not named particularly well, yes altitude of the camera from the ground plane. The function you refer to appears to take an X and Y position and returns the altitude of the land at that coordinate, this is useful but I want to know here how high above the land the camera is, raytracing with the actual geom is overkill and will take a lot of time, for this calculation a ground plane is sufficient. If there's a way to integrate the two functions that could be useful. I'll have another look at this.

Reply:
It's not super useful to call `glm::intersectRayPlane` with constant parameters.
For example, with plane_origin and plane_normal, we can simplify [`glm::intersectRayPlane`](https://github.com/g-truc/glm/blob/master/glm/gtx/intersect.inl#L5-L26) to be.
```c++
		if (glm::abs(rayDirection.y) > std::numeric_limits<float>::epsilon())
		{
			float intersectDistance = -_position.y / rayDirection .y;
			if (intersectDistance > 0.0f) {
				return intersectDistance;
			}
		}
		return 0.0f; // invalid value
```

In fact, that was in case you choose to make `rayDirection` variable. If it isn't, this is the function simplified

```c++
			if (_position.y > 0.0f) {
				return _position.y;
			}
            else
            {
                return 0.0f;  // invalid value
            }
```

Reply:
good point, this is just the y component of the camera position! over-complicating things..

Reply:
removed all references to this function in favor of:
`land.GetHeightAt(glm::vec2(_position.x, _position.z))`

Reply:
which is the actual land height from your previous function

Reply:
nope, actually none of that is necessary, just using `_position.y` is more than sufficient. some issues around this, fixing now..